src/plugins/workspace/public/components/data_source_association/association_data_source_modal.tsx

```diff
@@ -154,18 +166,28 @@ const convertConnectionsToOptions = ({
         return [];
       }
 
-      if (showDirectQueryConnections) {
-        if (!connection.relatedConnections || connection.relatedConnections.length === 0) {
-          return [];
+      if (connection.connectionType === DataSourceConnectionType.OpenSearchConnection) {
+        if (showDirectQueryConnections) {
+          if (!connection.relatedConnections || connection.relatedConnections.length === 0) {
+            return [connection];
```

Comment:
Could you add a comment on why we return [connection] when connection is empty in this PR?

Reply:
ok, Thanks for all the comments, all updated.

Reply:
Saw the new comment, if a open search connection has none child connections, will this still work if we set [connection]?

Reply:
yes, if a connection doesn't have related connections, it will just return [connection]. please check the attached video

https://github.com/user-attachments/assets/f668c5d3-bea1-4636-885f-dc2d80715e5e